pkg/sql/schemachanger/rel/database_entity_set.go

```diff
@@ -60,6 +60,32 @@ func (t *entitySet) insert(v interface{}, es entityStore) (int, error) {
 	for _, field := range ti.fields {
 		var val uintptr
 		switch {
+		case field.isSlice():
+			// In this case, we want to create slice members for each member in
+			// the slice and then be done with it.
+			fvi := field.value(vp)
+			if fvi == nil {
+				continue
+			}
+			fv := reflect.ValueOf(fvi)
+			if fv.Kind() != reflect.Slice {
+				return 0, errors.AssertionFailedf(
+					"expected a slice type, got %v", fv.Type(),
+				)
+			}
+			for i := 0; i < fv.Len(); i++ {
+				smpv := reflect.New(field.sliceMemberType.Elem())
+				smv := smpv.Elem()
+				smv.Field(sliceMemberSourceFieldIndex).Set(value)
```

Comment:
what about the other index we expanded with attributes `Attrs: []Attr{idx.Attrs[0], sliceSource, sliceIndex}` ?